### PR TITLE
Decompose process_attestations() into helper functions

### PR DESCRIPTION
**Pull request by pablodeymo**
## Motivation

  `process_attestations()` was 183 lines with validation, vote recording, justification, finalization, and serialization all inlined in a single
  function. This makes it hard to review each phase independently and to reason about which parts mutate state.

  ## Description

  Extract three logical phases into well-named helper functions:

  ### `is_valid_attestation()` — pure predicate, no mutation
  Consolidates the 6 consecutive `continue` guards into a single function returning `bool`. Checks:
  1. Source is already justified
  2. Target is not yet justified
  3. Neither root is zero hash
  4. Both checkpoints exist in `historical_block_hashes`
  5. Target slot > source slot
  6. Target slot is justifiable after the finalized slot

  ### `try_finalize()` — finalization attempt after justification
  Extracts the finalization block that was nested inside the supermajority `if` block, inside the attestation loop (three levels of nesting).
  Inverts the condition to use early return, removing one nesting level.

  ### `serialize_justifications()` — post-loop SSZ conversion
  Extracts the deterministic conversion from the in-memory `HashMap<H256, Vec<bool>>` vote structure back into SSZ-compatible
  `state.justifications_roots` and `state.justifications_validators`.

  ### What stays inline
  Vote recording and the supermajority check remain in the main loop because they're tightly coupled to the iteration (mutating `justifications` and
   `attestations_processed` on every pass).

  ### No behavior change
  - Public API unchanged (`process_block` → `process_attestations`)
  - All existing comments preserved
  - No logic reordering within each extracted block

  ## How to Test

  ```bash
  make fmt
  make lint
  make test
  cargo test -p ethlambda-blockchain --features skip-signature-verification --test forkchoice_spectests --release
  ```

  The forkchoice spectests (26 tests) exercise `process_attestations` end-to-end: justification, finalization, reorgs, and edge cases. All pass.

### CODE REVIEW

File: crates/blockchain/state_transition/src/lib.rs [hunk 1]
```diff
@@ -233,42 +233,7 @@ fn process_attestations(
...
-        }
-
-        // Ensure time flows forward
-        if target.slot <= source.slot {
-            continue;
-        }
-
-        // Ensure the target falls on a slot that can be justified after the finalized one.
-        if !slot_is_justifiable_after(target.slot, original_finalized_slot) {
+        if !is_valid_attestation(state, source, target, original_finalized_slot) {
```
[the remark this answers is not in the document]
Comment (pablodeymo) [reply crates/blockchain/state_transition/src/lib.rs]: done!
